### DISCUSSION TITLE
no-invalid-this in addEventListener anonymous function

### DISCUSSION THREAD

**Discussion by erosman** [eslint/eslint#17750] [Q&A] [answered]
Should there be an error?

```js
/* eslint no-invalid-this: "error" */
document.querySelector('select').addEventListener('change', function () {
  console.log(this.value);
});
```

**Comment by mdjermanovic** [comment, marked as answer]
Sorry for the lack of response.

This rule can't be aware of all APIs, so in these cases you'll need to use eslint-disable comments.

We also have a few in the eslint codebase, for example:

https://github.com/eslint/eslint/blob/90077199fe519953f9af8664bf947db4e4958514/tests/lib/cli.js#L87-L95
